File: solutions/cpp/secret-handshake/1/secret_handshake.cpp
#include "secret_handshake.h"
#include <algorithm>

using namespace std;

namespace secret_handshake {

    vector<string> commands(int cmd) {

        vector<string> handshake;
        
        if ((cmd & 0b00001) != 0)
            handshake.push_back("wink");

        if ((cmd & 0b00010) != 0)
            handshake.push_back("double blink");

        if ((cmd & 0b00100) != 0)
            handshake.push_back("close your eyes");

        if ((cmd & 0b01000) != 0)
            handshake.push_back("jump");

        if ((cmd & 0b10000) != 0)
            reverse(handshake.begin(), handshake.end());
        
        return handshake;
    }

    
}  // namespace secret_handshake


        

File: solutions/csharp/secret-handshake/1/SecretHandshake.cs
using System;
using System.Collections.Generic;

public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        var commands = new List<string>();

        if ((commandValue & 0b00001) != 0)
            commands.Add("wink");

        if ((commandValue & 0b00010) != 0)
            commands.Add("double blink");

        if ((commandValue & 0b00100) != 0)
            commands.Add("close your eyes");

        if ((commandValue & 0b01000) != 0)
            commands.Add("jump");

        if ((commandValue & 0b10000) != 0)
            commands.Reverse();

        return commands.ToArray();
    }
}
